software/contrib/noddy_holder.py

```diff
@@ -0,0 +1,162 @@
+from europi import *
+import machine
+from europi_script import EuroPiScript
+from time import ticks_diff, ticks_ms
+
+'''
+NoddyHolder
+author: Sean Bechhofer (github.com/seanbechhofer)
+date: 2022-03-26
+labels: sample & hold, trigger & hold
+
+Two channels of sample and hold/trigger and hold. 
+
+Digital in: Gate
+Analog in: CV
+
+Output 1: Gate
+Output 2: Sample and Hold based on Gate and CV
+Output 3: Trigger and Hold based on Gate and CV
+
+Output 4: Inverted Gate
+Output 5: Sample and Hold based on Inverted Gate and CV
+Output 6: Trigger and Hold based on Inverted Gate and CV
+
+digital_in: gate
+analog_in: cv
+
+knob_1: Not used
+knob_2: Not used
+
+button_1: Not used
+button_2: Not used
+
+output_1: gate 
+output_2: s&h
+output_3: t&h
+output_4: inverted gate
+output_5: s&h using inverted gate
+output_6: t&h using inverted gate
+
+'''
+# Version number
+
+VERSION="1.0"
+
+# Overclock the Pico for improved performance.
+machine.freq(250_000_000)
+
+class NoddyHolder(EuroPiScript):
+    def __init__(self):
+        self.gate = False
+        # Keep track of values for display, S&H and T&H for each
+        # channel
+        self.channel_1 = [0,-1]
+        self.channel_2 = [0,-1]
+        
+        # Triggered when button 1 is released
+        # Short press: 
+        # Long press: 
+        @b1.handler_falling
+        def b1Pressed():
+            if ticks_diff(ticks_ms(), b1.last_pressed()) >  300:
+                # long press
+                pass
+            else:
+                # short press
+                pass
+
+        # Triggered when button 2 is released.
+        # Short press: 
+        # Long press: 
+        @b2.handler_falling
+        def b2Pressed():
+            
+            if ticks_diff(ticks_ms(), b2.last_pressed()) >  300:
+                # long press
+                pass
+            else:
+                # short press
+                pass
```

Comment:
You can delete this section since it looks like buttons ended up not being used with this script.

Reply:
Fair point.